rdkit/Chem/UnitTestPandasTools.py

```diff
@@ -1,80 +1,207 @@
 from __future__ import print_function
+
+import gzip
 import os
+import shutil
+import tempfile
 import unittest
-
-from rdkit.six.moves import cStringIO as StringIO
-from rdkit import RDConfig
-
-from rdkit.Chem import PandasTools
-gotPandas = PandasTools.pd is not None
+import doctest
 
 import numpy
-import tempfile, shutil
-import gzip
 
-methane = """\
-Methane
-     RDKit
-
-  1  0  0  0  0  0  0  0  0  0999 V2000
-    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
-M  END
-> <prop1>
-12.34
-
-> <prop2>
-qwe
+from rdkit import RDConfig, rdBase, Chem
+from rdkit.Chem import PandasTools
+from rdkit.six import PY3
+from rdkit.six import StringIO, BytesIO
 
-$$$$
-"""
+# We make sure that we don't mess up the Mol methods for the rest of the tests
+PandasTools.UninstallPandasTools()
 
-peroxide = """\
-Peroxide
-     RDKit
 
-  2  1  0  0  0  0  0  0  0  0999 V2000
-    0.0000    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
-    0.0000    0.0000    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
-  1  2  1  0
-M  END
-> <prop2>
-rtz
+@unittest.skipIf(PandasTools.pd is None, 'Pandas not installed, skipping')
+class TestPandasTools(unittest.TestCase):
 
-> <prop3>
-yxcv
+  def __init__(self, methodName='runTest'):
+    self.df = getTestFrame()
+    self.df.index.name = 'IndexName'
+    super(TestPandasTools, self).__init__(methodName=methodName)
 
-$$$$
-"""
+  def setUp(self):
+    PandasTools.InstallPandasTools()
+    PandasTools.ChangeMoleculeRendering(renderer='PNG')
+    self._molRepresentation = PandasTools.molRepresentation
+    self._highlightSubstructures = PandasTools.highlightSubstructures
+
+  def tearDown(self):
+    PandasTools.molRepresentation = self._molRepresentation
+    PandasTools.highlightSubstructures = self._highlightSubstructures
+    PandasTools.UninstallPandasTools()
+
+  def testDoctest(self):
+    # We need to do it like this to ensure that default RDkit functionality is restored
+    suite = unittest.TestSuite()
+    suite.addTests(
+      doctest.DocTestSuite(PandasTools, optionflags=doctest.ELLIPSIS +
+                           doctest.NORMALIZE_WHITESPACE))
+    result = unittest.TestResult()
+    result = suite.run(result)
+    self.assertEqual(result.testsRun, 1)
+    self.assertEqual(len(result.errors), 0)
+    self.assertEqual(len(result.failures), 0)
```

Comment:
I don't particularly like this part, however running the doctests as done in other unittest files  wouldn't allow making and restoring the PandasTools modifications. It works, but it might be easier if we just move the doctests from the PandasTools.py to here.